goaviatrix/account.go

```diff
@@ -218,18 +219,38 @@ func (c *Client) CreateAWSSAccount(account *Account) error {
 	return c.PostFileAPI(params, files, DuplicateBasicCheck)
 }
 
-func (c *Client) GetAccount(account *Account) (*Account, error) {
+func (c *Client) ListAccounts() ([]Account, error) {
+	// If cached accounts are recent enough, return them
+	c.cacheMutex.Lock()
+	defer c.cacheMutex.Unlock()
+	if c.cachedAccounts != nil && time.Since(c.cacheTimestamp) < 10*time.Minute {
```

Comment:
if the cache is for one run of terraform it seems like we don't need to time-bound it -- it seems reasonable to say that each run of terraform should have a consistent view of accounts?

Reply:
Correct.  We can make it non-expiring, I just threw something in there to prove it out.
Each command should be stateless so the cache will be rebuilt for each command, (plan, apply etc...)  I only see this is way to force one call to happen per command.

Reply:
Removed expiry